src/mainboard/pcengines/apu2/mainboard.c

```diff
@@ -473,8 +473,11 @@ static int read_serial_from_nic(char *serial, size_t len)
 	uintptr_t bar10;
 	u32 mac_addr = 0;
 	int i;
+	if (check_pciereverse())
```

Comment:
any checks for apu4? or apu5?

Reply:
Why would it be needed?  If I understand correctly we want to calculate serial always from the same MAC of first NIC. After reversing order of PCIe devices this NIC changes its place independently from which apu it is.

Reply:
After reversing order this NIC is no longer first in the system, but I assumed that serial number should remain unchanged. Is it correct?

Reply:
Yes, the serial number must remain unchanged. I mean apu5 has onyl 2 NICs, so if you check for PCIe 2.3 it will hit a non-NIC device or nothing.

Reply:
It would hit a non-NIC without reverse PCIe order option enabled, but PCIe 2.3 is only checked when it is enabled, so when the first NIC is addressed as 2.3
I changed the address of the first NIC in another file, so here I just take into account this change.

Reply:
So when dev is not enabled we end up without a serial number?

Reply:
Yes, in this case the serial number would equal '0'. It worked in the same way previously. We could calculate the serial number from different NIC, if it is enabled, but to get the same serial number we would have to assume that we know the difference between their MAC addresses. Which, I guess, is the case, MAC addresses on the next NIC should be an incremented MAC of the previous NIC, but I'm not sure if this is always true.
I'm not sure if it is possible to try to enable the 1st NIC ones again at this point in code, but if it is possible I could try it too.